src/jsx-info.js

```diff
@@ -1,2 +1,63 @@
 #!/usr/bin/env node
-require("./cli").default();
+const {
+  directory,
+  gitignore,
+  files,
+  components,
+  report,
+  sort,
+  showProgress
+} = require("./cli");
+const parser = require("./parser");
+const Reporter = require("./reporter");
+const printer = require("./printer");
+const codeSource = require("./code-source");
+
+// Find Files
+if (showProgress) printer.printProgress();
+const filenames = codeSource.searchForFiles(files, gitignore, directory);
+
+// Scan Files
+const reporter = new Reporter(sort);
+const scanTasks = filenames.map(filename => {
+  return new Promise((res, rej) => {
+    if (showProgress) printer.printScanningFile(filename);
+
+    parser(codeSource.codeFromFile(filename), {
+      onlyComponents: components
+    })
+      .on("component", reporter.addComponent.bind(reporter))
+      .on("prop", reporter.addProp.bind(reporter))
+      .on("child", reporter.addChild.bind(reporter))
+      .on("finish", res)
+      .on("error", error => {
+        if (error instanceof SyntaxError) {
+          reporter.addParseError(filename, error);
+          res();
+        } else {
+          rej(error);
+        }
+      });
+  });
+});
+
+// Report
+Promise.all(scanTasks).then(() => {
```

Comment:
I don't think you want `Promise.all`, it fails if a single promise is rejected.

Reply:
Also, if we need to keep async stuff, you can use async functions instead of `.then` chaining